File: Selenide_Solid/src/test/java/com/ust/sdet/selenide/pages/LoginPage.java
package com.ust.sdet.selenide.pages;

import com.codeborne.selenide.SelenideElement;
import com.ust.sdet.selenide.support.SelenideConfig;

import static com.codeborne.selenide.Condition.visible;
import static com.codeborne.selenide.Selenide.*;
import static com.codeborne.selenide.Selenide.webdriver;
import static com.codeborne.selenide.WebDriverConditions.urlContaining;

public class LoginPage {

    private final SelenideElement email = $("#email");
    private final SelenideElement password = $("#password");
    private final SelenideElement signIn =
            $("#main-content section form button");
    private final SelenideElement loginPageTitle = $("#login-title");
    private final SelenideElement title = $("#page-title");
    private final SelenideElement signOut =
            $("#main-content section.hero div.hero-copy div button");

    public LoginPage() {
    }

    public LoginPage openSignPage() {
        open(SelenideConfig.loginUrl());
        loginPageTitle.shouldBe(visible);
        return this;
    }

    public LoginPage login(String email, String password) {
        this.email.setValue(email);
        this.password.setValue(password);
        signIn.click();

        webdriver().shouldHave(urlContaining("/home"));
        return this;
    }

}

File: UI/pages/LoginPage.ts
import {Page} from "@playwright/test"

export class LoginPage{

    page : Page;

    constructor(page: Page){
        this.page = page
    }

    async goto(){
        await this.page.goto('/login')
    }

    async login(email: string, password: string){
        await this.page.getByRole('link', { name: /Log in/i }).click();
        await this.page.getByRole('textbox', { name: /Email/i }).click();
        await this.page.getByRole('textbox', { name: /Email/i }).fill(email);
        await this.page.getByRole('textbox', { name: /Password/i }).click();
        await this.page.getByRole('textbox', { name: /Password/i }).fill(password);
        await this.page.getByRole('button', { name: /Sign in/i }).click();

    }

}``